### PR TITLE
People 194 notificatin email

### PR DESCRIPTION
**Pull request by DariuszMusielak**
# Ticket
https://netguru.atlassian.net/browse/PEOPLE-194

# Description
- first day every month we send email to users with unread notifications. 
- refactored specs
- added template for email
- added rake task and setup cron

### Before merge checklist
- [ ] CircleCI is passing
- [ ] Proper labels are set

### CODE REVIEW

File: app/mailers/user_mailer.rb [hunk 1]
```diff
@@ -19,6 +19,13 @@ def notify_admins_about_changes(user, current_user)
     mail(to: to, subject: "#{user.first_name} #{user.last_name} updated.")
   end
 
+  def notify_unread_notifications(user)
+    @unread_notifications_count = user.unread_notifications_count
+    @notification_text = 'notification'.pluralize(@unread_notifications_count)
```
Comment (marcinjak) [inline app/mailers/user_mailer.rb]: Do you use these variables outside the function? Maybe local variables (not instance variables) would be enough :)

Comment (DariuszMusielak) [inline app/mailers/user_mailer.rb]: Yeah, I use them in mail template

File: lib/tasks/mailer.rake [hunk 2]
```diff
@@ -30,6 +30,23 @@ namespace :mailer do
     end
   end
 
+  desc 'Email about users that have some unread notifications'
+  task users_with_unread_notifications: :environment do
+    unread_notifications_count = %((select count(*) FROM notifications WHERE notifications.receiver_id = users.id AND notifications.notification_status = 'unread'))
```
Comment (netguru-hound) [inline lib/tasks/mailer.rake]: Line is too long. [164/100]

File: app/views/user_mailer/notify_unread_notifications.html.haml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+%p
+  = "You have #{@unread_notifications_count} unread #{@notification_text}."
+  = link_to "Go to notifications page.", notifications_url
+
+%p{ style: 'display: inline-block; padding-top: 6px; border-top: 1px solid; font-size: 0.9em; margin-top: 10px !important;' }
+  We send this email first day of every month (only if) you have some unread notifications.
```
Comment (netguru-hound) [inline app/views/user_mailer/notify_unread_notifications.html.haml]: Line is too long. [91/80]

File: app/views/user_mailer/notify_unread_notifications.html.haml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+%p
+  = "You have #{@unread_notifications_count} unread #{@notification_text}."
+  = link_to "Go to notifications page.", notifications_url
+
+%p{ style: 'display: inline-block; padding-top: 6px; border-top: 1px solid; font-size: 0.9em; margin-top: 10px !important;' }
```
Comment (netguru-hound) [inline app/views/user_mailer/notify_unread_notifications.html.haml]: Line is too long. [125/80]

File: app/views/user_mailer/notify_unread_notifications.html.haml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+%p
+  = "You have #{@unread_notifications_count} unread #{@notification_text}."
```
Comment (netguru-hound) [inline app/views/user_mailer/notify_unread_notifications.html.haml]: `= "..."` should be rewritten as `...`

File: app/mailers/user_mailer.rb [hunk 6]
```diff
@@ -19,6 +19,13 @@ def notify_admins_about_changes(user, current_user)
...
+    subject = "#{user.first_name} #{user.last_name} has #{@unread_notifications_count} unread #{@notification_text}."
```
Comment (netguru-hound) [inline app/mailers/user_mailer.rb]: Line is too long. [117/100]

**Pull request merged**